yaptide/routes/user_routes.py

```diff
@@ -72,13 +79,46 @@ def get(user: UserModel):
                         'input_type': simulation.input_type,
                         'sim_type': simulation.sim_type
                     }
-                }
-                for simulation in simulations],
-            'page_count': pagination.pages,
-            'simulations_count': pagination.total,
+                } for simulation in simulations
+            ],
+            'page_count':
+            pagination.pages,
+            'simulations_count':
+            pagination.total,
         }
         return yaptide_response(message='User Simulations', code=200, content=result)
 
+    @staticmethod
+    @requires_auth()
+    def delete(user: UserModel):
+        """Method deleting simulation from database"""
+        schema = UserSimulations.APIParametersSchema()
+        errors: dict[str, list[str]] = schema.validate(request.args)
+        if errors:
+            return error_validation_response(content=errors)
+        params_dict: dict = schema.load(request.args)
+
+        job_id = params_dict['job_id']
+        simulation = fetch_simulation_by_job_id(job_id)
+
+        if simulation is None:
+            return yaptide_response(message=f'Simulation with job_id={job_id} do not exist', code=404)
+
+        # Simulation has to be completed/cancelled before deleting it.
+        if simulation.job_state in (EntityState.UNKNOWN.value, EntityState.PENDING.value, EntityState.RUNNING.value):
+            return yaptide_response(message=f'''Simulation with job_id={job_id} is currently running.
+                  Please cancel simulation or wait for it to finish''',
+                                    code=403)
+
+        estimators: list[EstimatorModel] = fetch_estimators_by_sim_id(sim_id=simulation.id)
+        if len(estimators) > 0:
+            for estimator in estimators:
+                delete_object_from_db(estimator)
+
+        delete_object_from_db(simulation)
```

Comment:
```suggestion
        estimators: list[EstimatorModel] = fetch_estimators_by_sim_id(sim_id=simulation.id)
        if len(estimators) > 0:
            for estimator in estimators:
                delete_object_from_db(estimator, make_commit=False)

        delete_object_from_db(simulation)
```

what do you think of delaying commit until last deletion ? that would speedup the whole process

Reply:
Agree, it's a good idea to delete both objects in one db transaction